### ISSUE TITLE
Multiple Token Regex Search Not Working

### ISSUE THREAD

**Issue by ParmeetSingh** [OpenGrok/OpenGrok#1598]
Hey,

I am using OpenGrok 1.0. I am able to search for a regex query like /Gen[e]ric/. This successfully gives results.

However, I am not able to search for /Gen[e]ric Master/ although the phrase Generic Master exists in my codebase.
Does OpenGrok not support multiple token regex queries?

Can you please help?

**Comment by tarzanek** [comment]
http://lucene.apache.org/core/6_5_1/core/org/apache/lucene/util/automaton/RegExp.html?is-external=true

is what is used underneath ... 
so I think this should work ... will need to investigate deeper

**Comment by ParmeetSingh** [comment]
Hey,

I am using a whitespace(default) tokenizer. So, do you think Lucene will be able to perform regex search across index terms(even if using Automaton)?
I tried indexing using Line-Tokenizer after which I was able to perform multi-token regex search.

Not sure for whitespace(default) tokenizer.Can you please clarify?

**Comment by tarzanek** [comment]
Well I'd hope all of {OpenGrok tokenizers would work by default ... if they don't (which from your mail looks like that), then we need to figure it out and fix it ... obviously if you already figured it out, we'd be happy to accept a patch

**Comment by idodeclare** [comment]
> Does OpenGrok not support multiple token regex queries?

No, unfortunately Lucene regex query expressions operate on single tokens. In fact, they must match an entire token. E.g., if I revise your example as `/Gen[e]r/`, that would not match a term "Generic". Likewise `/Gen[e]ric Master/` would not match any single token in the OpenGrok index.

**Comment by tarzanek** [comment]
so /Gen[e]ric/ Master  should be the closest one one can get ... ?

**Comment by idodeclare** [comment]
> so /Gen[e]ric/ Master should be the closest one one can get ... ?

That's right, and I don't know of a way to use regex with proximity searches like you can do with static phrase queries like `"generic master"`.
